plugins/vppv2/puntplugin/README.md

```diff
@@ -1,29 +1,177 @@
+
+**1. Using Key-value database:** put proto-modeled data to the database under the correct key for punt to host configuration. If the database is the ETCD, it is possible to use [vpp-agent-ctl](../../../cmd/vpp-agent-ctlv2) utility. Let's assume that we want to configure a punt to host with socket registration (with default agent label `vpp1`):
+
+```
+vpp-agent-ctl -put "/vnf-agent/vpp1/vpp/config/v2/punt/tohost/l3/IPv4/l4/UDP/port/9000" json/punt-to-host.json
+```  
+
+The more simple way is to use the configuration example directly in the `vpp-agent-ctl`. The example contains pre-defined data which can be put to the ETCD with a single command. Commands are different for a punt to host alone, or registered via socket:
+
+```
+vpp-agent-ctl -puntr        // simple punt to host
+vpp-agent-ctl -rsocket      // register punt to host via socket
+```
+
+To remove/deregister data, use:
+
+```
+vpp-agent-ctl -puntd
+vpp-agent-ctl -dsocket
+```
+
+**2. Using REST** not yet implemented
+
+**2. Using GRPC:** not yet implemented
+
+### Limitations
+<a name="pths-limit"></a>
+
+Current limitations for a punt to host:
+* The UDP configuration cannot be shown (or even configured) in the VPP cli.
+* The VPP does not provide API to dump configuration, which takes the vpp-agent the opportunity to read existing entries and may case issues with resync.
+* Although the vpp-agent supports the TCP protocol as the L4 protocol to filter incoming traffic, the current VPP version don't.
+* Configured punt to host entry cannot be removed since the VPP does not support this option. The attempt to do so exits with an error.
+
+Current limitations for a punt to host via unix domain socket:
+* The configuration cannot be shown (or even configured) in the VPP cli.
+* The vpp-agent cannot read registered entries since the VPP does not provide an API to dump.
+* The VPP startup config punt section requires unix domain socket path defined. The limitation is that only one path can be defined at a time.
+
+### Known issues
+<a name="pths-issues"></a>
+
+* VPP issue: if the Unix domain socket path is defined in the startup config, the path has to exist, otherwise the VPP fails to start. The file itself can be created by the VPP.
+
+## IP redirect
+
+Defined as the IP punt, IP redirect allows a traffic matching given IP protocol to be punted to the defined TX interface and next hop IP address. All those fields have to be defined in the northbound proto-modeled data. Optionally, the RX interface can be also defined as an input filter.  
+
+### Model
+<a name="ipr-model"></a>
+
+IP redirect is defined as `IpRedirect` object in the generated proto model.
+
+The key has a following format:
+```
+vpp/config/v2/punt/ipredirect/l3/<L3_protocol>/tx/<tx_interface_name>"
+```
+
+L3 protocol is defined as `string` value (transformed to numeric in VPP API call). The table is the same as before. Supported values:
+
+|L3 protocol |Numeric |
+|:----------:|:------:|
+|UNDEFINED_L3|0       |
```

Comment:
This shouldn't be required in the documentation. It's defined to be able to compare it to zero value, but the actual value is not valid.